### PR TITLE
feat: [CI-10068]: Added os and arch in docker delegate command

### PR DESCRIPTION
**Pull request by devkimittal**
cg call

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55230)

### CODE REVIEW

File: 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      + "${docker_deploy_mode_string}"
+      + "  -e NEXT_GEN=\"true\" \\\n"
+      + "  -e DELEGATE_TYPE=\"DOCKER\" \\\n"
+      + "  -e ACCOUNT_ID=${account_id} \\\n"
+      + "  -e DELEGATE_TOKEN=${token} \\\n"
+      + "  -e DELEGATE_TAGS=\"linux-amd64\" \\\n"
+      + "  -e LOG_STREAMING_SERVICE_URL=${manager_url}/log-service/ \\\n"
+      + "  -e MANAGER_HOST_AND_PORT=${manager_url} ${image}";
+
+  public static String getCommand(StringSubstitutor substitute, String os, String arch) {
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java:34]: looks like arch is not used here. remove ?

Comment (devkimittal) [reply 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java:34]: later we have plans to pass it in arguments and change the command accordingly

File: 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  private static final String DOCKER_COMMAND = "docker run --cpus=1 --memory=2g \\\n"
+      + "  -e DELEGATE_NAME=docker-delegate \\\n"
+      + "${docker_deploy_mode_string}"
+      + "  -e NEXT_GEN=\"true\" \\\n"
+      + "  -e DELEGATE_TYPE=\"DOCKER\" \\\n"
+      + "  -e ACCOUNT_ID=${account_id} \\\n"
+      + "  -e DELEGATE_TOKEN=${token} \\\n"
+      + "  -e LOG_STREAMING_SERVICE_URL=${manager_url}/log-service/ \\\n"
+      + "  -e MANAGER_HOST_AND_PORT=${manager_url} ${image}";
+  private static final String DOCKER_COMMAND_LINUX = "docker run -d --cpus=1 --memory=2g --net=host \\\n"
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java:23]: it's better not to create another constant DOCKER_COMMAND_LINUX, 'cuz future developers will likely miss changes in either of the places
can you refer to the "${docker_deploy_mode_string}" approach , and do something similar ?

File: 400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java [hunk 3]
```diff
@@ -807,12 +808,13 @@ public Response downloadDelegateValuesYaml(@Context HttpServletRequest request,
   @ExceptionMetered
   public RestResponse<Map<String, String>> getInstallationCommand(@Context HttpServletRequest request,
       @QueryParam("accountId") @NotEmpty String accountId, @QueryParam("orgId") String orgId,
-      @QueryParam("projectId") String projectId, @QueryParam("commandType") @NotEmpty String commandType)
-      throws IOException {
+      @QueryParam("projectId") String projectId, @QueryParam("commandType") @NotEmpty String commandType,
+      @QueryParam("os") String os, @QueryParam("arch") String arch) throws IOException {
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java:812]: i don't see a place where arch is used. can we remove ?

File: 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java (hunk 2 above)
Comment (devkimittal) [reply 400-rest/src/main/java/io/harness/delegate/utils/DockerDelegateInstallationCommandHelper.java]: removed

**Review by rajatharanganath — APPROVED** [review]

**Pull request merged**